### DISCUSSION TITLE
How do I add custom properties when uploading multiple files at once?

### DISCUSSION THREAD

**Discussion by morris14** [spatie/laravel-medialibrary#2379] [Q&A] [answered]
I have a vue component (using pro), and am trying to upload multiple images at once using addFromMediaLibraryRequest:

```php
$project
    ->addFromMediaLibraryRequest($request->attachments)
    ->withCustomProperties([
        'author_id' => auth()->id(),
        'author_name' => auth()->user()->name,
    ])
    ->toMediaCollection('attachments');
```

But I keep getting the error: `Spatie\MediaLibraryPro\Dto\PendingMediaItem::Spatie\MediaLibraryPro\Dto\{closure}(): Argument #1 ($customProperty) must be of type string, array given`, I don't get this when just using `addMedia()` with a single upload. How would I go about attaching the logged in users info to custom property on all images when uploading multiple images?

Many thanks!

**Comment by flowbru** [comment]
Did you find a solution to this? I need the exact same thing...

**Comment by morris14** [comment, marked as answer]
@flowbru just had a look back at the project I had the issues with and looks like I didn't find a direct fix. I landed with this approach though:

```php
if (auth()->user()->isInTeam($client->id) || auth()->user()->isAdmin) {
    $project
        ->addFromMediaLibraryRequest($request->attachments)
        ->toMediaCollection('attachments');

    foreach ($request->attachments as $image) {
        $mediaItem = Media::where('uuid', $image['uuid'])->first();
        $mediaItem->setCustomProperty('author', [
            'id' => auth()->id(),
            'name' => auth()->user()->name
        ]);
        $mediaItem->save();
    }
}
```

Not particularly efficient, but does the job!

Hope this helps!

**Reply by flowbru** [reply]
Thank you very much @morris14 for your answer and the code!
I had something similar in mind but hoped, there is an easier way.

**Reply by erikn69** [reply]
Maybe better performance with `whereIn`
```php
Media::whereIn('uuid', array_column($request->attachments, 'uuid'))->get()->each(function($mediaItem){
    $mediaItem->setCustomProperty('author', [
        'id' => auth()->id(),
        'name' => auth()->user()->name
    ]);
    $mediaItem->save();
});
```
